CMakeLists.txt

```diff
@@ -195,11 +195,7 @@ function(RELATIVE_PROTOBUF_GENERATE_CPP NAME SRCS HDRS ROOT_DIR DEPEND)
       file(MAKE_DIRECTORY "${OUTPUT_PROTO_DIR}")
     endif()
 
-    if("${PYTHON_EXECUTABLE}" STREQUAL "")
-      set(_python_exe "python")
-    else()
-      set(_python_exe "${PYTHON_EXECUTABLE}")
-    endif()
+    find_package(PythonInterp REQUIRED)
```

Comment:
```suggestion
    if (${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.12.0")
        # find_package Python has replaced PythonInterp and PythonLibs since cmake 3.12
        find_package(Python ${PY_VERSION} REQUIRED)
    else()
        find_package(PythonInterp ${PY_VERSION} REQUIRED)
        find_package(PythonLibs ${PY_VERSION} REQUIRED)
    endif()
```

Reply:
I suggest using `find_package(Python)` for `cmake >=3.12.0`.

`find_package(PythonInterp)` and `find_package(PythonLibs)` has been deprecated since cmake 3.12.0
https://cmake.org/cmake/help/v3.12/module/FindPythonInterp.html
Even though they can work now, it would be better to use the latest API to prevent failures in the future.

BTW, in line#446, there is a same function doing the same thing. We should try to merge these things. Thanks!

Reply:
I think it should be `find_package(Python ${PY_VERSION} REQUIRED COMPONENTS Interpreter)` for recent cmake.

I actually considered using that when writing the patch, but opted for not doing so for two reasons:
* The minimum cmake version that is declared in the `CMakeLists.txt` does not allow doing so.
* The deprecated variant is already elsewhere.

Therefore it is consistent to also use it here. Using the current syntax is a good thing, but only if it allows removing the deprecated code. Keeping both seems to be the worst of both ways. Given cmake's approach to backwards compatibility, it does not seem likely for it to remove the deprecated variant anytime soon.

For these reasons, I'd leave it as is.

Merging it with the later invocation conceptually makes sense. However, these were separate prior to this PR (i.e. not a regression) and we're trying to fix a build failure here. Cleaning this up risks deferring this bug fix. In case you insist, where would you place the shared `find_package` call? Around line 80?

Reply:
Thank you for the explanation. My viewpoint is we should use the latest API whenever applicable. Actually the original deprecated variant (line#446) in CMakeLists.txt might cause an error on Windows https://github.com/onnx/onnx/pull/1631.

Anyway it's a suggestion and I understand your concern. If you don't have bandwidth to handle this, I will try it in my PR. Thank you for catching this issue.

(In case you are interested, around line 80 is good.)